recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
+            # Set native system header dir to ${{sysroot}}/usr/include to
+            # isolate installation from the system as much as possible
+            tc.configure_args.append("--with-native-system-header-dir=/usr/include")
+            tc.make_args.append("BOOT_LDFLAGS=-Wl,-headerpad_max_install_names")
+        tc.generate()
+
+        # Don't use AutotoolsDeps here - deps are passed directly in configure_args.
+        # Using AutotoolsDeps causes the compiler tests to fail by erroneously adding
+        # additional $LIBS to the test compilation
 
-    @property
-    def _make_args(self):
-        if self.settings.os == "Macos":
-            return ["BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"]
-        return []
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def build(self):
         # If building on x86_64, change the default directory name for 64-bit libraries to "lib":
-        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
-        tools.replace_in_file(os.path.join(self.source_folder,
-                                           self._source_subfolder, "gcc", "config", "i386", "t-linux64"),
-                              "m64=../lib64", "m64=../lib", strict=False)
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "gcc", "config", "i386", "t-linux64"),
+            "m64=../lib64",
+            "m64=../lib",
+            strict=False,
+        )
+
         # Ensure correct install names when linking against libgcc_s;
         # see discussion in https://github.com/Homebrew/legacy-homebrew/pull/34303
-        tools.replace_in_file(os.path.join(self.source_folder,
-                                           self._source_subfolder, "libgcc", "config", "t-slibgcc-darwin"),
-                              "@shlib_slibdir@", libdir, strict=False)
-        autotools = self._configure_autotools()
-        autotools.make(args=self._make_args)
-
-    def package_id(self):
-        del self.info.settings.compiler
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "libgcc", "config", "t-slibgcc-darwin"),
+            "@shlib_slibdir@",
+            os.path.join(self.package_folder, "lib"),
+            strict=False,
+        )
+
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
-        if self.settings.build_type == "Debug":
-            autotools.install(args=self._make_args)
-        else:
-            autotools.make(args=["install-strip"] + self._make_args)
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.remove_files_by_mask(self.package_folder, "*.la")
-        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+        autotools = Autotools(self)
+        autotools.make(target="install-strip")
```

Comment:
I believe the preferred choice is to now use the autotools install method given that it accepts a target override and automatically sets DESTDIR correctly, which is needed for handling the package_folder correctly in Conan 2.0:

> autotools.install(target="install-strip")